crates/nomination/src/lib.rs

```diff
@@ -143,12 +144,11 @@ pub mod pallet {
         #[transactional]
         pub fn opt_in_to_nomination(
             origin: OriginFor<T>,
-            collateral_currency: CurrencyId<T>,
-            wrapped_currency: CurrencyId<T>,
+            currency_pair: DefaultVaultCurrencyPair<T>,
         ) -> DispatchResultWithPostInfo {
             ext::security::ensure_parachain_status_running::<T>()?;
             let account_id = ensure_signed(origin)?;
-            let vault_id = VaultId::new(account_id, collateral_currency, wrapped_currency);
+            let vault_id = VaultId::new(account_id, currency_pair.collateral, currency_pair.wrapped);
```

Comment:
Is it worth changing this constructor to take the currency pair directly?

Reply:
I would prefer to merge now and to consider doing this in the follow up pr